app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/NavigationOverlayFragment.kt

```diff
@@ -271,6 +276,18 @@ class NavigationOverlayFragment : Fragment() {
         ).map { channel -> channel.forwardRemoveEventsToRepo() }
     }
 
+    private fun observeToolbarState(): Disposable {
+        return navigationOverlayViewModel.sessionState
+                .subscribe { state ->
+                    navUrlInput.nextFocusUpId = when {
+                        state.backEnabled -> R.id.navButtonBack
+                        state.forwardEnabled -> R.id.navButtonForward
+                        state.currentUrl != URLs.APP_URL_HOME -> R.id.navButtonReload
+                        else -> R.id.turboButton
+                    }
```

Comment:
There are a few issues here.  The biggest one is that we're duplicating logic.  If the code in `ToolbarVM` changed, this would fall out of sync and cause weird focus bugs.  Another is that this logic shouldn't live in the View at all, because it isn't its responsibility.  Let's work together on Monday to find a better place for this to live.